packages/cozy-realtime/src/RetryManager.js

```diff
@@ -0,0 +1,142 @@
+import MicroEE from 'microee'
+
+import {
+  maxWaitBetweenRetries,
+  baseWaitAfterFirstFailure,
+  timeBeforeSuccessfull,
+  raiseErrorAfterAttempts
+} from './config'
+
+class RetryManager {
+  constructor() {
+    this.reset()
+    this.onSuccess = this.onSuccess.bind(this)
+    this.onFailure = this.onFailure.bind(this)
+    this.onSuccessAcknowledged = this.onSuccessAcknowledged.bind(this)
+  }
+
+  /**
+   * On a success
+   *
+   * If `timeBeforeSuccessfull` if not null, it will wait this amount of
+   * time without errors before acknowleding a success
+   */
+  onSuccess() {
+    if (timeBeforeSuccessfull > 0) {
+      this.startSuccessWatcher(
+        this.onSuccessAcknowledged,
+        timeBeforeSuccessfull
+      )
+    } else {
+      this.onSuccessAcknowledged()
+    }
+  }
+
+  /**
+   * Acknowledge a success
+   *
+   * @private
+   */
+  onSuccessAcknowledged() {
+    this.emit('success')
+    this.reset()
+  }
+
+  /**
+   * Clears the success watcher
+   *
+   * @private
+   */
+  clearSuccessWatcher() {
+    if (this.successWatcher) {
+      global.clearTimeout(this.successWatcher)
+      this.successWatcher = null
+    }
+  }
+
+  /**
+   * Starts the success watcher - acknowledges the success if not interrupted after some time
```

Comment:
I have trouble understanding why we need this ? Could the docstring be expanded with a rationale behind this ?

Reply:
:+1: